tasks/go_deps.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import os
+from collections.abc import Iterable
+
+from invoke.context import Context
+from invoke.exceptions import Exit
+from invoke.tasks import task
+
+from tasks.build_tags import get_default_build_tags
+from tasks.diff import BINARIES
+from tasks.flavor import AgentFlavor
+from tasks.go import GOARCH_MAPPING, GOOS_MAPPING
+from tasks.libs.common.color import color_message
+from tasks.libs.common.datadog_api import create_gauge, send_metrics
+
+
+def compute_count_metric(
+    ctx: Context,
+    build: str,
+    flavor: AgentFlavor,
+    platform: str,
+    arch: str,
+    entrypoint: str,
+    timestamp: int | None = None,
+    extra_tags: Iterable[str] = (),
+):
+    """
+    Compute a metric representing the number of Go dependencies of the given build/flavor/platform/arch,
+    and one with only dependencies outside of the agent repository.
+    """
+
+    if not timestamp:
+        timestamp = int(datetime.datetime.now(datetime.UTC).timestamp())
+
+    goos, goarch = GOOS_MAPPING[platform], GOARCH_MAPPING[arch]
+    tags = [
+        f"build:{build}",
+        f"flavor:{flavor.name}",
+        f"os:{goos}",
+        f"arch:{goarch}",
+    ]
+    tags.extend(extra_tags)
```

Comment:
Nit. For ease of reading I would move this tags definitions to line 59. Since in this section of the function we are collecting the dependencies and also use tags it might create a bit of confusion.